manifest.md

```diff
@@ -175,6 +175,14 @@ If the `config.mediaType` is set to the empty value, the `artifactType` MUST be
 If the artifact does not need layers, a single layer SHOULD be included with a non-zero size.
 The suggested content for an unused `layers` array is the [empty descriptor](#guidance-for-an-empty-descriptor).
 
+The design of the artifact depends on what content is being packaged with the artifact.
+The decision tree below MAY be used to structure your artifact:
+
+1. Does the artifact consist of at least one file or blob? If yes, continue to 2. If no, specify the `artifactType`, and set the `config` and a single `layers` element to the empty descriptor value.
+2. Does the artifact have additional JSON formatted metadata as configuration? If yes, continue to 3. If no, specify the `artifactType`, include the artifact in the `layers`, and set `config` to the empty descriptor value.
```

Comment:
For non-image artifact, does it allow this usage, the layer descriptor in the layers pointing to another manifest, i.e, the media type in the layer is "application/vnd.oci.image.manifest.v1+json".

In current image-spec/manifest.md description, it does not set the restriction on this usage.

Reply:
I believe that's a separate issue/question from this PR, so it would be better to raise that on its own for visibility.

My own take is that the spec doesn't say the media type allowed for the layer, only what container runtimes should support. However, I think you'll encounter issues pushing that to a registry and handled by generic tools because the layers field content is treated as an opaque blob, not necessarily processed to extract content like a manifest. Blobs are pushed to a separate API on the registry, registries may reject a manifest if all blobs are not pushed, and a manifest pushed as a blob may not be processed by the registry to retain embedded content from the garbage collector.

If you want a manifest and references other manifests, then you likely want an OCI Index.